esteid/signing/views.py

```diff
@@ -123,58 +118,90 @@ def finish_session(self, request: "RequestType", *args, **kwargs):
             do_cleanup = False
             return JsonResponse({"status": self.Status.PENDING, **e.data}, status=e.status)
 
-        except EsteidError as e:
-            return JsonResponse({"status": self.Status.ERROR, **e.get_user_error()}, status=e.status)
-
-        except Exception:
-            logger.exception("Failed to finalize signature.")
-            return JsonResponse(
-                {"status": self.Status.ERROR, "error": "Internal error"}, status=HTTPStatus.INTERNAL_SERVER_ERROR
-            )
-
         finally:
             if do_cleanup:
                 signer.cleanup()
 
         return self.get_success_response(*args, **kwargs)
 
-
-class SignViewRestMixin(SignViewMixin):
-    """
-    To be used with rest-framework's APIView
-    """
+    def report_error(self, e: EsteidError):
+        return JsonResponse({"status": self.Status.ERROR, **e.get_user_error()}, status=e.status)
 
     def post(self, request, *args, **kwargs):
         """
         Handles session start requests
         """
-        return self.start_session(request, *args, **kwargs)
+        try:
```

Comment:
:thinking: A lot of duplicate code in `post` and `patch` - on other hand, maybe this is the case where we want to be verbose and explicit.

Reply:
I wouldn't call it _a lot_, it's just exception handling. Moving that out into a separate entity would definitely make it less intuitive.